### PR TITLE
[EFR32] Link the door lock relock behaviour with the app UI

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
The door lock cluster implementation mostly left the callback to be implemented by the app.

However, the relock callback is implemented directly in the cluster server implementation and changes the attribute Lock state directly. Make that callback weak and let an application reimplement it if needed.

Re-Implement the callback in the efr32 application implementation so it updates our UI directly (LCD and LED

### CODE REVIEW

File: src/app/clusters/door-lock-server/door-lock-server.cpp [hunk 1]
```diff
@@ -3751,8 +3751,9 @@ void MatterDoorLockClusterServerAttributeChangedCallback(const app::ConcreteAttr
 // Timer callbacks
 // =============================================================================
 
-void emberAfPluginDoorLockOnAutoRelock(chip::EndpointId endpointId)
+void __attribute__((weak)) emberAfPluginDoorLockOnAutoRelock(chip::EndpointId endpointId)
```
Comment (bzbarsky-apple) [inline src/app/clusters/door-lock-server/door-lock-server.cpp:3754]: Wait, why is this needed?  Shouldn't there be a callback into the app-provided delegate from `DoorLockServer::Instance().SetLockState` here if the app needs to be informed this is happening?  If there is no such callback, that sounds like the real problem....

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]

File: src/app/clusters/door-lock-server/door-lock-server.cpp (hunk 1 above)
Comment (jmartinez-silabs) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:3754]: I was trying to maintain how the `door-lock-server.h` seems to be implemented, which is a series of callback stubs for command handling to be implemented by the application. There is no direct callback for the SetLockState if that is what you ask or at least I don't see it. I could use the `MatterPostAttributeChangeCallback` to reflect the lock state change in the app but that is not how it was implemented for any of the other events that change the door lock attributes.

I thought `emberAfPluginDoorLockOnAutoRelock` should be a callback that can be implemented by the app like all the other stubs listed in `src/app/clusters/door-lock-server/door-lock-server-callback.cpp`

Comment (bzbarsky-apple) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:3754]: You're right; I had forgotten how door lock is set up.

If we follow how the rest of door lock works, there should be two functions here:

1. Some function (name TBD) that's set up as the timer handler and does what `emberAfPluginDoorLockOnAutoRelock does right now and calls `emberAfPluginDoorLockOnAutoRelock`.
2. `emberAfPluginDoorLockOnAutoRelock`, which is a weak symbol in `door-lock-server-callback.cpp` which just returns void and does nothing and can be overridden by the application to do other additional things as needed.

In terms of this PR, what that means is that the "set event control inactive" and "set lock state" bits would stay in the door lock cluster code and the app would just need to do the `LockMgr().InitiateAction(AppEvent::kEventType_Lock, LockManager::LOCK_ACTION);` bit...  Or we could consider moving the `SetLockState` call into the app too, but note that in this PR the app call is getting the `DlOperationSource` wrong, which is a very good argument for not just leaving it to the app...

Comment (jmartinez-silabs) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:3754]: alright, that sounds good. I will update the PR and create this second weak callback for app implementation

**Review by selissia — APPROVED** [review]

File: src/app/clusters/door-lock-server/door-lock-server.cpp [hunk 2]
```diff
@@ -69,7 +69,7 @@ class DoorLockClusterFabricDelegate : public chip::FabricTable::Delegate
 };
 static DoorLockClusterFabricDelegate gFabricDelegate;
 
-void emberAfPluginDoorLockOnAutoRelock(chip::EndpointId endpointId);
+void DoorLockOnAutoRelockCallback(chip::EndpointId endpointId);
```
Comment (bzbarsky-apple) [inline src/app/clusters/door-lock-server/door-lock-server.cpp:72]:
```suggestion
static void DoorLockOnAutoRelockCallback(chip::EndpointId endpointId);
```

But actually, is there a reason to not just make this a static member function of DoorLockServer?  Then you won't need the friend declaration either....

**Review by bzbarsky-apple — APPROVED** [review]

File: src/app/clusters/door-lock-server/door-lock-server.cpp (hunk 2 above)
Comment (jmartinez-silabs) [reply src/app/clusters/door-lock-server/door-lock-server.cpp:72]: `AutolockEvent` was private from that context so the way it was placed it needed to be a friend of the door lock class. I previously just renamed the old function but you are right. Now, I changed this to make `DoorLockOnAutoRelockCallback` a static private member of the Door lock class and remove the local declaration. This is a better representation of what the callback is
